dhall/src/Dhall/Syntax.hs

```diff
@@ -1092,87 +918,16 @@ pathCharacter c =
-denote  NaturalEven           = NaturalEven
-denote  NaturalOdd            = NaturalOdd
-denote  NaturalToInteger      = NaturalToInteger
-denote  NaturalShow           = NaturalShow
-denote  NaturalSubtract       = NaturalSubtract
-denote (NaturalPlus a b     ) = NaturalPlus (denote a) (denote b)
-denote (NaturalTimes a b    ) = NaturalTimes (denote a) (denote b)
-denote  Integer               = Integer
-denote (IntegerLit a        ) = IntegerLit a
-denote  IntegerClamp          = IntegerClamp
-denote  IntegerNegate         = IntegerNegate
-denote  IntegerShow           = IntegerShow
-denote  IntegerToDouble       = IntegerToDouble
-denote  Double                = Double
-denote (DoubleLit a         ) = DoubleLit a
-denote  DoubleShow            = DoubleShow
-denote  Text                  = Text
-denote (TextLit (Chunks a b)) = TextLit (Chunks (fmap (fmap denote) a) b)
-denote (TextAppend a b      ) = TextAppend (denote a) (denote b)
-denote  TextShow              = TextShow
-denote  List                  = List
-denote (ListLit a b         ) = ListLit (fmap denote a) (fmap denote b)
-denote (ListAppend a b      ) = ListAppend (denote a) (denote b)
-denote  ListBuild             = ListBuild
-denote  ListFold              = ListFold
-denote  ListLength            = ListLength
-denote  ListHead              = ListHead
-denote  ListLast              = ListLast
-denote  ListIndexed           = ListIndexed
-denote  ListReverse           = ListReverse
-denote  Optional              = Optional
-denote (Some a              ) = Some (denote a)
-denote  None                  = None
-denote  OptionalFold          = OptionalFold
-denote  OptionalBuild         = OptionalBuild
-denote (Record a            ) = Record (fmap denote a)
-denote (RecordLit a         ) = RecordLit (fmap denote a)
-denote (Union a             ) = Union (fmap (fmap denote) a)
-denote (Combine _ b c       ) = Combine Nothing (denote b) (denote c)
-denote (CombineTypes a b    ) = CombineTypes (denote a) (denote b)
-denote (Prefer a b c        ) = Prefer a' (denote b) (denote c)
-  where
-    a' = case a of
-        PreferFromSource     -> PreferFromSource
-        PreferFromWith e     -> PreferFromWith (denote e)
-        PreferFromCompletion -> PreferFromCompletion
-denote (RecordCompletion a b) = RecordCompletion (denote a) (denote b)
-denote (Merge a b c         ) = Merge (denote a) (denote b) (fmap denote c)
-denote (ToMap a b           ) = ToMap (denote a) (fmap denote b)
-denote (Field a b           ) = Field (denote a) b
-denote (Project a b         ) = Project (denote a) (fmap denote b)
-denote (Assert a            ) = Assert (denote a)
-denote (Equivalent a b      ) = Equivalent (denote a) (denote b)
-denote (With a b c          ) = With (denote a) b (denote c)
-denote (ImportAlt a b       ) = ImportAlt (denote a) (denote b)
-denote (Embed a             ) = Embed a
+denote (Embed a      ) = Embed a
+denote (Combine _ b c) = Combine Nothing (denote b) (denote c)
```

Comment:
If we remove the "origin info" from `Combine` here, should we do something similar with `Prefer`?

Reply:
Possibly, although right now I'm just trying to make the change as behavior-preserving as possible and the previous implementation of `denote` did not strip the `PreferAnnotation`